src/sourmash/search.py

```diff
@@ -195,19 +316,12 @@ def search_databases_with_flat_query(query, databases, **kwargs):
     results.sort(key=lambda x: -x[0])
 
     x = []
-    ksize = query.minhash.ksize
+    search_scaled = query.minhash.scaled # is this true?
```

Comment:
You are right to ask, and I don't know the answer with 100% confidence. I think it will be the max across _all_ matches, plus query? But ... hmm. For gather I know we may increase scaled in the middle of the search if we hit a high-scaled signature, but I'm not sure where we test that (...I hope we do...)

Reply:
(The problem with taking the max of the scaled for each (query, match) pair is that we return the original, unmodified signature/minhash to these routines, not the downsampled one.)

Reply:
The relevant tests are the ones that have text like `WARNING: final scaled was 10000, vs query scaled of 500` in them -

* `tests/test_sourmash.py::test_gather_lca_db`
* `tests/test_sourmash.py::test_gather_query_downsample`

but it turns out neither of these tests test _multiple_ rounds of downsampling. Sigh.

See new PR, https://github.com/sourmash-bio/sourmash/pull/1956, which you can merge into this 'un to get some tests for both gather and prefetch. I think the test code should be reasonably clear as to what's going on...?

Reply:
using this test code in `latest` branch, for the prefetch CSV output, I get the max for each pair:
```
% csvtk cut -f scaled xyz.csv                      scaled                                                                          750                                                                             600
1000
650
```
and for the `gather` CSV output, I get
```
% csvtk cut -f scaled xyz.csv
scaled
600
650
750
1000
```
which is a little bit surprising (not sure why they are monotonically increasing, maybe I should adjust test?)